### DISCUSSION TITLE
Advice for a Newbie

### DISCUSSION THREAD

**Discussion by Ampurcell** [cortexrd/Knack-Toolkit-Library#108] [Q&A]
Hello, 

I am an absolute novice with much of this coding and syntax. 
I have added a bunch of keywords successfully, and they have been major game changers. 
I was hoping to discover solutions for a few more items, but I don't know if they are new concepts or if I misread the keyword descriptions. 

**1. Open a Link in a new page from a _GRID_ view.** 
    - I didn't think the _oln= and the _ols work in a grid view. Or do they and I am reading wrong? The link would be a connection link populated from the record.  

**2. I would like to hide a column formatted for grouping in a _GRID_ view.** 
    - I have some grids I wanted to lock 1st position sorting using a field connection and could only manage to get it to function the way I wanted if I added the connected field and made it a group. Otherwise, the other sorting would override it. _hc didn't work, and I didn't want to remove grouping bc the sorting would change whenever I used another column. But it is taking up real estate. 

**3. Text view max character in a _GRID_ view.** 
    - I may have seen a similar request. But I don't want long text wrapping or expanding the column in a Grid view.        
    Just the first 15 characters or something. Then, you would need to click the cell (inline editing is active) to see or edit the full text. 

Thank you for reading. 
Feel free to let me know if you can help or provide any additional direction. 

Alisha P

**Comment by cortexrd** [comment]
Hi Alisha,

Glad to read that you enjoy using the KTL and that it helps solving some of your problems.

About your questions:

1. Open a Link in a new page from a GRID view.
_I didn't think the _oln= and the _ols work in a grid view. Or do they and I am reading wrong? The link would be a connection link populated from the record._

No, indeed the two keywords only work in Rich Text Views, as documented.

But from the tests I did a moment ago - just to be sure, when I add a link field type in a grid and I click on it, by default it opens that link in the same page (same as _ols).  Now, in the field settings, you also have an option (check box) "Open in new window" that does the same as _oln.

You may be interested to know that if you do a **Ctrl+Click** on a link, it will open the link in a new tab in your current browser, and **Shift+Click** will open the link in a new browser.

With all these options... I wonder if you may have some other specific use cases, or just something that I'm missing?


2. I would like to hide a column formatted for grouping in a GRID view.

If you want to hide a column that has the group attribute, then you need to use another method.  This is simply because the header text disappears and the _hc code uses that as a trigger.

So what I recommend is using the **_cls** keyword (add/remove classes)
See here:  [https://github.com/cortexrd/Knack-Toolkit-Library/wiki/Keywords#_clsclass1-class2](url)

There are several built-in KTL classes that are used internally, but also available to the user.  See the **KTL.css** file here to see all of them: [https://github.com/cortexrd/Knack-Toolkit-Library/blob/dev/KTL.css](url)

You need this one: **ktlDisplayNone**

Here are a few examples on how to use this keyword in your grid views:

`_cls=[ktlDisplayNone], [ktlTarget, $('.kn-table-group')]`
...to hide all groups.

`_cls=[ktlDisplayNone], [ktlTarget, $('.kn-group-level-1')]`
...to hide only group of level 1.

`_cls=[ktlDisplayNone], [ktlTarget, $('.kn-group-level-2')]`
...to hide only group of level 2.

`_cls=[ktlDisplayNone], [ktlTarget, $('.kn-group-level-1, .kn-group-level-3')]`
...to hide only group of level 1 and 3.



3. Text view max character in a GRID view.
_I may have seen a similar request. But I don't want long text wrapping or expanding the column in a Grid view.
Just the first 15 characters or something. Then, you would need to click the cell (inline editing is active) to see or edit the full text._

This is how you do it:
- In the grid view, select the columns that you want to restrain and set their width to 200px in this example.
- Add this keyword to you grid:
`_style=[white-space: nowrap; overflow: hidden; text-overflow: ellipsis; max-width: 200px;], [ktlTarget, $('td span')]`

About `[ktlTarget, $('td span')]`
This will be applied to all fields in your table.  Up to you to decide if this is desirable.

Of course, you have several options:

`[ktlTarget, $('td.field_1 span')]`
... if you want the trunking to be applied to a single field.

`[ktlTarget, $('td.field_1 span, td.field_2 span')]`
... if you want the trunking to be applied to fields 1 and 2 only.

Hope all this help,
Norm

**Comment by cortexrd** [comment]
Alisha,

I've created a new keyword just for you!

**_trk**  _Truncate Cell Text_

See documentation here:  [https://github.com/cortexrd/Knack-Toolkit-Library/wiki/Keywords#_trk](url)

No need for _style, ktlTarget and width values anymore.
It will adapt automatically to the Builder's individual column width settings.

You must have **KTL v0.15.19** to see it.  Do a **Ctrl+F5** to force an update of your browser if you don't see it.

Enjoy,
Norm

**Comment by Ampurcell** [comment]
Thank you so much!! You are literally the best. I am slammed today, but I will go through it tonight!

**Comment by Ampurcell** [comment]
Hi Again! 

I finally got a chance to try it out. Thanks so much again. You are the best! 

1. I know I can use ctrl and right-click. But I constantly forget. And links for connections in the grid don't have the option to open on a new page.
![Screenshot 2023-10-06 at 10 03 37 AM](https://github.com/cortexrd/Knack-Toolkit-Library/assets/141195753/3d87e136-3d55-4c91-b0aa-2a345575ffde)

2.  WORKED PERFECT!!! Took me a minute. But that's EXACTLY what I needed. Thank you. 

3. Love the new keyword! Thank you so much. But is it possible to have a symbol or something simpler than "view more". It takes up characters. lol

**Reply by cortexrd** [reply]
Hmmm... I don't see the "**view more**" text.  All I see are ellipses.  Got a screenshot?

Here's mine:

![image](https://github.com/cortexrd/Knack-Toolkit-Library/assets/42851359/abf2c2dc-3b64-4e9a-9f35-dccea6a1388a)

**Comment by Ampurcell** [comment]
Oh! Do you have a donations page? All of these items are amazing! I don't have a alot of money, but I would like to give you something.

**Reply by cortexrd** [reply]
That's something I wanted to do for a while.

DONE!

It's on the main page next to **"Sponsor this project"**

And direct link here:  [https://www.paypal.com/donate/?hosted_button_id=5JU6CKPCMMDZL](url)

Note: all donations will be given to collaborators who develop new features, debug, provide support, or help in any way.

Thanks a lot,
Norm

**Comment by cortexrd** [comment]
> 1. I know I can use ctrl and right-click. But I constantly forget. And links for connections in the grid don't have the option to open on a new page.

I've added support for the **_oln** keyword in all view types that can have links.
Just at it to your view and **all links** will always open in a new tab.

> 2. WORKED PERFECT!!! Took me a minute. But that's EXACTLY what I needed. Thank you.
Very happy to read that.

> 3. Love the new keyword! Thank you so much. But is it possible to have a symbol or something simpler than "view more". It takes up characters. lol
I'd need a screen grab.  I don't see that.

Norm

**Comment by cortexrd** [comment]
### Update on _oln keyword

I've fixed several bugs with the new **_oln** support in views.  They are now only supported in **Grids** and **Details** views.  Not across the board because it caused too much problems.  I will support more upon request as we go.

Sorry for the trouble,
Norm
